nixos/common/flake.nix

```diff
@@ -42,6 +57,9 @@ in
     };
   };
   config = lib.mkIf (cfg.flake != null) {
+    nix.settings.nix-path = lib.mkIf (cfg.flake.inputs ? nixpkgs) "nixpkgs=${cfg.flake.inputs.nixpkgs}";
+    nix.settings.flake-registry = lib.mkDefault flake-registry;
```

Comment:
Not sure we want this by default because this will upload nixpkgs to all kinds of machines, which can overhead and slow to upload. There is also a proper nixos option for this: https://github.com/Mic92/dotfiles/blob/9e2f4683cdf6a942e8ffc65ed3aa5ef27aaa4652/nixos/modules/pinned-registry.nix#L3

Reply:
Just now the module was using builtins.unsafeDiscardStringContext

Reply:
Also potentially Nixos has something like this now as well?